src/boot/java/io/github/brassmc/brassloader/boot/discovery/ModDiscovery.java

```diff
@@ -0,0 +1,260 @@
+package io.github.brassmc.brassloader.boot.discovery;
+
+import com.google.auto.service.AutoService;
+import cpw.mods.jarhandling.SecureJar;
+import cpw.mods.modlauncher.api.IEnvironment;
+import cpw.mods.modlauncher.api.IModuleLayerManager;
+import cpw.mods.modlauncher.api.ITransformationService;
+import cpw.mods.modlauncher.api.ITransformer;
+import io.github.brassmc.brassloader.boot.mods.ModContainer;
+import org.apache.commons.validator.routines.UrlValidator;
+import org.hjson.JsonObject;
+import org.hjson.JsonValue;
+import org.hjson.ParseException;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+@AutoService(ITransformationService.class)
+public class ModDiscovery implements ITransformationService {
+    public static final List<ModContainer> MODS = new ArrayList<>();
+
+    private static final Path MODS_FOLDER = Path.of("mods");
+
+    @Override
+    public @NotNull String name() {
+        return "brass:moddiscovery";
+    }
+
+    @Override
+    public void initialize(IEnvironment environment) {
+
+    }
+
+    @Override
+    public List<Resource> completeScan(IModuleLayerManager layerManager) {
+        try {
+            final List<SecureJar> mods = new ArrayList<>();
+
+            Files.walkFileTree(MODS_FOLDER, new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    try {
+                        // Read jar
+                        SecureJar secureJar = SecureJar.from(file);
+                        mods.add(secureJar);
+
+                        Path metadata = secureJar.getPath("mod.hjson");
+                        InputStream stream;
+                        if (Files.exists(metadata))
+                            stream = Files.newInputStream(metadata);
+                        else
+                            return FileVisitResult.CONTINUE;
```

Comment:
```suggestion
                        if (Files.notExists(metadata)) return FileVisitResult.CONTINUE;
                        final var data = Files.readString(metadata);
```
And adapt the code to use the string.